pkg/registry/apis/example/dummy_storage.go

```diff
@@ -0,0 +1,121 @@
+package example
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	example "github.com/grafana/grafana/pkg/apis/example/v0alpha1"
+	grafanarequest "github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
+	grafanaregistry "github.com/grafana/grafana/pkg/services/grafana-apiserver/registry/generic"
+)
+
+var (
+	_ rest.Storage              = (*dummyStorage)(nil)
+	_ rest.Scoper               = (*dummyStorage)(nil)
+	_ rest.SingularNameProvider = (*dummyStorage)(nil)
+	_ rest.Getter               = (*dummyStorage)(nil)
+	_ rest.Lister               = (*dummyStorage)(nil)
+)
+
+type dummyStorage struct {
+	Store             *genericregistry.Store
+	names             []string
+	creationTimestamp metav1.Time
+
+	Sub *DummySubresourceREST
+}
+
+func newDummyStorage(gv schema.GroupVersion, scheme *runtime.Scheme, names ...string) *dummyStorage {
+	strategy := grafanaregistry.NewStrategy(scheme)
+	store := &genericregistry.Store{
+		NewFunc:                   func() runtime.Object { return &example.DummyResource{} },     // getter not supported
+		NewListFunc:               func() runtime.Object { return &example.DummyResourceList{} }, // both list and get return the same thing
+		PredicateFunc:             grafanaregistry.Matcher,
+		DefaultQualifiedResource:  gv.WithResource("dummy").GroupResource(),
+		SingularQualifiedResource: gv.WithResource("dummy").GroupResource(),
+		CreateStrategy:            strategy,
+		UpdateStrategy:            strategy,
+		DeleteStrategy:            strategy,
+	}
+	store.TableConvertor = rest.NewDefaultTableConvertor(store.DefaultQualifiedResource)
```

Comment:
It seems like we could pull this out into a separate function, pass the store into the constructor and keep it as a private member.

Reply:
good questions -- I'll explore more.  This was the result of copying patterns that work (and failing on many that did not).  It seems like deep in k8s it is checking if the `Store` property is the same --- but like many things with k8s, learned more since thinking I learned that and am quite possibly wrong :)